rubin_scheduler/scheduler/surveys/scripted_surveys.py

```diff
@@ -137,14 +137,15 @@ def add_observation(self, observation, indx=None, **kwargs):
                     detailer.add_observation(observation, **kwargs)
                 self.reward_checked = False
 
-                key = observation["scheduler_note"][0] + observation["band"][0]
-                try:
+                indx = np.where(
+                    (self.obs_wanted["scheduler_note"] == observation["scheduler_note"][0])
+                    & (self.obs_wanted["observed"] == False)
+                )[0]
```

Comment:
Could you make this split the scheduler note by ',' and match the case / name / cycle (i). / and then band, while also letting alternative bands match?